src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java

```diff
@@ -407,18 +405,20 @@ public static void flushBuildsWithNoBCList() {
     // innerDeleteEventToJenkinsJobRun is the actual delete logic at the heart
     // of deleteEventToJenkinsJobRun
     // that is either in a sync block or not based on the presence of a BC uid
-    private static synchronized void innerDeleteEventToJenkinsJobRun(
+    private static void innerDeleteEventToJenkinsJobRun(
             final Build build) throws Exception {
         final WorkflowJob job = getJobFromBuild(build);
         if (job != null) {
-            ACL.impersonate(ACL.SYSTEM,
-                    new NotReallyRoleSensitiveCallable<Void, Exception>() {
-                        @Override
-                        public Void call() throws Exception {
-                            cancelBuild(job, build, true);
-                            return null;
-                        }
-                    });
+            synchronized (job) {
```

Comment:
Actually, I think you can get rid of this synchronized as well .... the sync's in `deleteEventToJenkinsJobRun` should be sufficient, as I think the the call to this method in that method's for loop isn't going to occur.